src/MoBi.Presentation/DTO/RenameExpressionProfileDTO.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using OSPSuite.Assets;
+using OSPSuite.Core.Domain;
+using OSPSuite.Presentation.DTO;
+using OSPSuite.Utility.Validation;
+
+namespace MoBi.Presentation.DTO
+{
+   public class RenameExpressionProfileDTO : ValidatableDTO
+   {
+      private string _moleculeName;
+      private string _category;
+      private string _species;
+      private IReadOnlyList<string> _prohibitedNames;
+
+      public RenameExpressionProfileDTO()
+      {
+         Rules.Add(RenameExpressionProfileDTORules.UniqueName);
+         Rules.AddRange(RenameExpressionProfileDTORules.NonEmptyFieldRules());
+      }
+
+      public string Type { get; set; }
+
+      public string Name => Constants.ContainerName.ExpressionProfileName(MoleculeName, Species, Category);
+
+      public string MoleculeName
+      {
+         get => _moleculeName;
+         set
+         {
+            _moleculeName = value;
+            OnPropertyChanged(() => Name);
+            OnPropertyChanged(() => MoleculeName);
+         }
+      }
+
+      public string Category
+      {
+         get => _category;
+         set
+         {
+            _category = value;
+            OnPropertyChanged(() => Name);
+            OnPropertyChanged(() => Category);
+         }
+      }
+
+      public string Species
+      {
+         get => _species;
+         set
+         {
+            _species = value;
+            OnPropertyChanged(() => Name);
+            OnPropertyChanged(() => Species);
+         }
+      }
+
+      private static class RenameExpressionProfileDTORules
+      {
+         public static IEnumerable<IBusinessRule> NonEmptyFieldRules()
+         {
+            yield return createNonEmptyRule(item => item.Species, Assets.AppConstants.Validation.SpeciesCannotBeEmpty);
+            yield return createNonEmptyRule(item => item.Category, Assets.AppConstants.Validation.CategoryCannotBeEmpty);
+            yield return createNonEmptyRule(item => item.MoleculeName, Assets.AppConstants.Validation.MoleculeNameCannotBeEmpty);
+         }
+
+         private static IBusinessRule createNonEmptyRule(Expression<Func<RenameExpressionProfileDTO, string>> propertyToCheck, string errorCaption)
```

Comment:
change those method to use a get so that it's created only once and not every time